doc/api/inspector.md

```diff
@@ -108,6 +108,49 @@ with an error. [`session.connect()`] will need to be called to be able to send
 messages again. Reconnected session will lose all inspector state, such as
 enabled agents or configured breakpoints.
 
+## inspector.ws
+
+The Inspector can be interacted with via a websocket-based protocol. These APIs
+allow programatic control of the websocket, equivalent to use `--inspect*` CLI
+options.
+
+### ws.listen()
+
+Start the websocket listener.
+
+### ws.unlisten()
+
+Stop the websocket listener.
+
+### ws.id()
+
+Return the UUID string of the listener, or undefined if the listener is not
+started.
+
+### ws.host([newHost])
```

Comment:
I am not sure ws should be a singleton.

Implementation-wise we can listen on multiple ports. Also, we were considering a pipe transport in addition to the WS transport for scenarios like subprocesses.

So, maybe expose in API support for multiple "servers"? Then you would start a server on a different port and shutdown the one that was running before.